hedera-cryptography/hedera-cryptography-tss/src/main/java/com/hedera/cryptography/tss/api/TssParticipantDirectory.java

```diff
@@ -0,0 +1,360 @@
+         */
+        @NonNull
+        public Builder withThreshold(final int threshold) {
+            if (threshold <= 0) {
+                throw new IllegalArgumentException("Invalid threshold: " + threshold);
+            }
+            this.threshold = threshold;
+            return this;
+        }
+
+        /**
+         * Adds a participant entry to the builder.
+         *
+         * @param participantId the participant unique {@link Integer} representation
+         * @param numberOfShares the number of shares
+         * @param tssEncryptionPublicKey the pairing public key used to encrypt tss share portions designated to the participant represented by this entry
+         * @return the builder instance
+         * @throws IllegalArgumentException if participantId was previously added.
+         */
+        @NonNull
+        public Builder withParticipant(
+                final int participantId,
+                final int numberOfShares,
+                @NonNull final PairingPublicKey tssEncryptionPublicKey) {
+            if (participantEntries.containsKey(participantId))
+                throw new IllegalArgumentException(
+                        "Participant with id " + participantId + " was previously added to the directory");
+
+            participantEntries.put(participantId, new ParticipantEntry(numberOfShares, tssEncryptionPublicKey));
+            return this;
+        }
+
+        /**
+         * Builds and returns a {@link TssParticipantDirectory} instance based on the provided entries and schema.
+         *
+         * @param schema the signature schema
+         * @return the constructed ParticipantDirectory instance
+         * @throws NullPointerException if schema is null
+         * @throws IllegalStateException if there is no entry for the current participant
+         * @throws IllegalStateException if there are no configured participants
+         * @throws IllegalStateException if the threshold value is higher than the total shares
+         */
+        @NonNull
+        public TssParticipantDirectory build(@NonNull final SignatureSchema schema) {
+            Objects.requireNonNull(schema, "Schema must not be null");
+
+            if (isNull(selfEntry)) {
+                throw new IllegalStateException("There should be an entry for the current participant");
+            }
+
+            if (participantEntries.isEmpty()) {
+                throw new IllegalStateException("There should be at least one participant in the protocol");
+            }
+
+            if (!participantEntries.containsKey(selfEntry.participantId())) {
+                throw new IllegalStateException(
+                        "The participant list does not contain a reference to the current participant");
+            }
+
+            // Get the total number of shares of to distribute in the protocol
+            int totalShares = participantEntries.values().stream()
```

Comment:
`totalShares` may be `final`

Reply:
sorry have to reconfigure the ide so it makes all variable finals automatically